app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt

```diff
@@ -60,6 +64,35 @@ class LoginFragment : Fragment() {
             hideSoftKeyboard(context, rootView)
         }
 
+        if (BuildConfig.FLAVOR == "playStore") {
+
+            smartAuthViewModel.requestCredentials(SmartAuthUtil.getCredentialsClient(requireActivity()))
+
+            smartAuthViewModel.id
+                .nonNull()
+                .observe(viewLifecycleOwner, Observer {
+                    email.text = SpannableStringBuilder(it)
+                })
+
+            smartAuthViewModel.password
+                .nonNull()
+                .observe(viewLifecycleOwner, Observer {
+                    password.text = SpannableStringBuilder(it)
+                })
+
+            smartAuthViewModel.resolvableApiException.nonNull().observe(viewLifecycleOwner, Observer {
+                if (smartAuthViewModel.requestCode.value != null) {
+                    SmartAuthUtil.handleResolvableApiException(
+                        it, requireActivity(), smartAuthViewModel.requestCode.value!!)
```

Comment:
Do not use non-null assertions

Reply:
In latest commit: As `resolvableApiException` and `requestCode` are always updated together, I have put them together in a `Pair` in the ViewModel and the null case is thus avoided.